### PR TITLE
[build] fix jackson conflicts among cdc connectors

### LINKED ISSUE #2983: [Bug] Jackson version conflicts among MySQL/TiDB/MongoDB connectors
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Flink version

1.14

### Flink CDC version

3.0.0

### Database and its version

MySQL 5.7

### Minimal reproduce step

Put `flink-sql-connector-tidb-cdc-3.0.0.jar`, `flink-sql-connector-mysql-cdc-3.0.0.jar` and `flink-sql-connector-mongodb-cdc-3.0.0.jar` into lib folder at Flink home (Flink 1.14 in my case), then create a MySQL CDC table and submit a simple query like `select * from mysql_cdc_table`, there would be an error:

`java.io.InvalidClassException: com.ververica.cdc.connectors.shaded.com.fasterxml.jackson.databind.cfg.MapperConfig; incompatible types for field _mapperFeatures`

It's caused by inconsistent jackson versions of the CDC connectors:

- `flink-sql-connector-mongodb-cdc-3.0.0.jar`: 2.10.2
- `flink-sql-connector-mysql-cdc-3.0.0.jar`: 2.13.2
- `flink-sql-connector-tidb-cdc-3.0.0.jar`: 2.12.3

### What did you expect to see?

All CDC connectors use the same jackson version.

### What did you see instead?

Inconsistent jackson versions among the CDC connectors which conflicts with each other.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by link3280**
As titled.

close #2983

### CODE REVIEW

**Comment by link3280** [comment]
@leonardBang The failed tests seem unrelated. Would you kindly trigger a re-run for the failed tests?

**Comment by link3280** [comment]
ping @leonardBang

File: pom.xml [hunk 1]
```diff
@@ -95,6 +96,36 @@ under the License.
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (Shawn-Hx) [inline pom.xml:100]: Thanks for your contribution. I think using [jackson-bom](https://github.com/FasterXML/jackson-bom) here is more clear.

Comment (link3280) [reply pom.xml:100]: Good point. Thanks a lot for your input!

**Comment by link3280** [comment]
ping @leonardBang @PatrickRen

**Review by leonardBang — APPROVED** [review]
Thanks @link3280 for the contribution and @Shawn-Hx for the review work, LGTM

**Pull request merged**